File: 1.5/ariprog/ariprog.cpp
/*
ID: giliev91
LANG: C++
TASK: ariprog
*/
#include <iostream>
#include <fstream>
using namespace std;

int main()
{
    int N, M;
    ofstream fout("ariprog.out");
    ifstream fin("ariprog.in");
    fin >> N >> M;

    int max_bisq = M * M * 2 + 1;
    bool *is_bisq = new bool[max_bisq];
    fill(is_bisq, is_bisq + sizeof(is_bisq), false);

    for (int i = 0; i <= M; i++)
        for (int j = i; j <= M; j++)
            is_bisq[i * i + j * j] = true;

    int max_a = 0;
    int *bisq = new int[max_bisq];
    for (int i = 0; i < max_bisq; i++)
        if (is_bisq[i])
            bisq[max_a++] = i;

    int total = 0;
    int max_b = max_bisq / (N - 1) + 2;
    for (int b = 1; b <= max_b; b += 1)
        for (int a = 0; a < max_a; a++)
        {
            int x = bisq[a];
            int i = 0;
            for (i = 0; i < N; i++)
            {
                if (x >= max_bisq || !is_bisq[x])
                    break;
                x += b;
            }

            if (i == N)
            {
                fout << bisq[a] << " " << b << "\n";
                total++;
            }
        }

    if (total == 0)
        fout << "NONE\n";
    return 0;
}

File: 1.5/ariprog/ariprog.py
"""
ID: giliev91
LANG: PYTHON3
TASK: ariprog
"""

with open("ariprog.in", "r") as f:
    N = int(f.readline().strip())
    M = int(f.readline().strip())

bisq = set()
for i in range(M + 1):
    for j in range(i, M + 1):
        bisq.add(i**2 + j**2)


seq = []
for b in range(1, (2 * M**2) // (N - 1) + 2):
    for a in bisq:
        x = a
        for i in range(N):
            if x not in bisq:
                break
            x += b
        else:
            seq.append((a, b))


with open("ariprog.out", "w") as f:
    if seq:
        for s in seq:
            f.write("%d %d\n" % s)
    else:
        f.write("NONE\n")
